File: 20A40/cir-villalobos/completos/e34/Administrador.py
from Empleado import Empleado

class Administrador(Empleado):
    depto = "N/A"
    bono = 0

    def __init__(self,nombreC, salarioC, fechaNacC, deptoC, bonoC):
        self.nombre = nombreC
        self.salario = salarioC
        self.fechaNac = fechaNacC
        self.depto = deptoC
        self.bono = bonoC

    def setDepto(self,dept):
        if(not(len(dept) > 0)):    return False
        self.depto = dept
        return True
    
    def setBono(self,bono):
        if(not(bono >= 0)):    return False
        self.bono = bono
        return True

    def getDepto(self) : return self.depto
    def getBono(self)  : return self.bono

    def toString(self):
        concat =  (super().toString() + "\tDepartamento: " 
        + self.depto + "\tBono: " + str(self.bono))
        return concat

File: 20A40/cir-Enrique/Ejercicio 34/Administrador.java
public class Administrador extends Empleado {
    private String depto;
    private double bono;

    public Administrador(String nombreC, double salarioC, String fechaNacC, String deptoC, double bonoC) {
        super(nombreC, salarioC, fechaNacC);
        this.depto = deptoC;
        this.bono = bonoC;
    }

    public boolean setDepto(String dept) {
        if (dept.length() == 0)
            return false;
        this.depto = dept;
        return true;
    }
    
    public boolean setBono(double bono) {
        if (bono < 0)
            return false;
        this.bono = bono;
        return true;
    }


    public String getDepto() {
        return this.depto;
    }

    public double getBono() {
        return this.bono;
    }

    
    public String toString() {
        String concat = (super.toString() + "           " + this.depto + "          " + this.bono);
        return concat;
    }
}